mirgecom/mpi.py

```diff
@@ -151,6 +151,12 @@ def wrapped_func(*args, **kwargs):
 
         _check_gpu_oversubscription()
 
-        func(*args, **kwargs)
+        # Change this to 'if 1:' in order to debug mirgecom with pudb on rank 0.
+        # Connect to pudb on rank 0 with 'telnet 127.0.0.1 6899'.
+        if 0:
+            from pudb.remote import debug_remote_on_single_rank
+            debug_remote_on_single_rank(MPI.COMM_WORLD, 0, func, *args, **kwargs)
```

Comment:
I'm not sure if there is a better way to enable debugging. Unfortunately, this can not be done in the driver, as the code above prevents access to an MPI communicator.

```python
if "mpi4py.MPI" in sys.modules:
        raise RuntimeError("mpi4py.MPI imported before designated MPI entry "
                           "point. Check for prior imports.")
```

Reply:
Could you add something like:

```python
def pudb_remote_on_single_rank(func):
    @wraps(func)
    def wrapped_func(*args, **kwargs):
        # pylint: disable=import-error
        from pudb.remote import debug_remote_on_single_rank
        from mpi4py import MPI
        debug_remote_on_single_rank(MPI.COMM_WORLD, 0, func, *args, **kwargs)

    return wrapped_func
```

and then in the driver do:

```python
@mpi_entry_point
@pudb_remote_on_single_rank
def main(...):
```
?

Reply:
This looks better

Reply:
0cf62c1